### PR TITLE
Etherum test update and added basic devp2p tests

### PR DESCRIPTION
**Pull request by SebastianDremo**

### CODE REVIEW

File: src/Nethermind/Nethermind.Network.Test/Discovery/NodeLifecycleManagerTests.cs [hunk 1]
```diff
@@ -40,25 +41,35 @@ public class NodeLifecycleManagerTests
...
         private int _port = 1;
         private string _host = "192.168.1.27";
 
         [SetUp]
         public void Initialize()
         {
+            _discoveryManagerMock = Substitute.For<IDiscoveryManager>();
+            _discoveryConfigMock = Substitute.For<IDiscoveryConfig>();
+            
+
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.Network.Test/Discovery/NodeLifecycleManagerTests.cs:67]: double enter -> to removal

File: src/Nethermind/Nethermind.Network.Test/Discovery/NodeLifecycleManagerTests.cs [hunk 2]
```diff
@@ -78,20 +89,56 @@ public void Initialize()
...
             var evictionManager = new EvictionManager(_nodeTable, logManager);
+            _evictionManagerMock = Substitute.For<IEvictionManager>();
             var lifecycleFactory = new NodeLifecycleManagerFactory(_nodeTable, new DiscoveryMessageFactory(_timestamper), evictionManager, new NodeStatsManager(statsConfig, logManager), discoveryConfig, logManager);
 
             _udpClient = Substitute.For<IMessageSender>();
 
             var discoveryDb = new SimpleFilePublicKeyDb("Test","test", logManager);
             _discoveryManager = new DiscoveryManager(lifecycleFactory, _nodeTable, new NetworkStorage(discoveryDb, logManager), discoveryConfig, logManager, _ipResolverMock);
             _discoveryManager.MessageSender = _udpClient;
+
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.Network.Test/Discovery/NodeLifecycleManagerTests.cs:105]: another useless whitespace, dunno why I created so many of them :D

File: src/Nethermind/Nethermind.Network.Test/Discovery/DiscoveryManagerTests.cs [hunk 3]
```diff
@@ -150,7 +150,7 @@ public void MemoryTest()
                 for (int b = 0; b < 255; b++)
                 {
                     INodeLifecycleManager manager = _discoveryManager.GetNodeLifecycleManager(new Node($"{a}.{b}.1.1", 8000));
-                    manager.SendPing();
+                    manager.SendPingAsync();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network.Test/Discovery/DiscoveryManagerTests.cs:153]: fire and forget? Or you missed await?

Comment (SebastianDremo) [reply src/Nethermind/Nethermind.Network.Test/Discovery/DiscoveryManagerTests.cs:153]: +1 to forget 😄

**Pull request merged**